### PR TITLE
Add extra .223 magazines

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

```SUMMARY: Content "Add extra .223 magazines"```
#### Purpose of change

Add extra magazines that actually exist for .223 weapons; especially high-capacity double-drum magazines; added pretty much all magazines about which I could find some info that they really exist. Also change a few existing magazines a bit to be more consistent with the newly-added ones, and made G36 use non-STANAG magazines.
#### Describe the solution

JSON editing.
#### Describe alternatives you've considered

None.
#### Testing

Made sure that game starts up with no errors with those additions.
#### Additional context

Weight, volume, price and spawn frequency for new magazines are for the most part eyeballed; any suggestions on those are welcome.

### CODE REVIEW

**Comment by carmakazi** [comment]
I would consider making the 20 round magazine the default for the Mini-14, I don't think anyone really uses the flush 5 rounders unless they're legally obligated to for state gun laws or maybe hunting.  Would maybe give the rifle some game life.  (It looks like it's just a matter of putting it first in the compatible mags list?)  Feel free to disagree.

If you're doing the G36, maybe change the AUG to its own mags as well?  They're plastic and come in 10, 30, and 42 rounds according to Steyr USA's website.  I don't believe any sort of drum exists.

What does the mag_compact flag do?  I see it missing from a couple of magazines that look like they should have it, namely the smaller Ruger ones and the standard G36 one.

**Comment by anothersimulacrum** [comment]
`MAG_COMPACT` means it can be stored in some ammo pouches.

File: data/json/items/magazine/223.json [hunk 1]
```diff
@@ -80,26 +149,188 @@
...
+    "capacity": 40,
+    "reliability": 7,
+    "flags": [ "MAG_COMPACT" ]
+  },
+  {
+    "id": "stanag42",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 42-round magazine" },
+    "description": "An increased-capacity 40-round box magazine for use with STANAG compatible rifles.",
```
Comment (ymber) [inline data/json/items/magazine/223.json:174]: Description doesn't match `capacity`.

**Comment by Rail-Runner** [comment]
> I would consider making the 20 round magazine the default for the Mini-14

Done.

> If you're doing the G36, maybe change the AUG to its own mags as well?

Done. I added a 100-round double drum magazine for AUG too, though; apparently it does exist but is discontinued, so likely to be rare.
https://www.galatiinternational.com/9mm-steyr-aug-c-mag-magazine-100-round-clear-drum-black-pouch-beta.html
https://en.wikipedia.org/wiki/Beta_C-Mag

> What does the mag_compact flag do? I see it missing from a couple of magazines that look like they should have it, namely the smaller Ruger ones and the standard G36 one.

They are supposed to inherit that flag from the items they're copied from; I'll double-check if they really do, though.

File: data/json/items/magazine/223.json [hunk 2]
```diff
@@ -80,26 +149,228 @@
...
+    "ammo_type": [ "223", "300blk" ],
+    "capacity": 40,
+    "reliability": 7,
+    "flags": [ "MAG_COMPACT" ]
+  },
+  {
+    "id": "stanag42",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "STANAG 42-round magazine" },
```
Comment (tenmillimaster) [inline data/json/items/magazine/223.json:173]: Where are you finding/what are you basing these magazines off of?

**Comment by tenmillimaster** [comment]
Could you provide clearer references to what you're basing the magazines off of? Mostly the 42, 50, 60, 100, and 150 round stanags.

**Comment by Rail-Runner** [comment]
> Could you provide clearer references to what you're basing the magazines off of? Mostly the 42, 50, 60, 100, and 150 round stanags.

42-round STANAG: https://gunmagwarehouse.com/promag-ar-15-223-5-56-42-round-magazine.html
50-round STANAG (though I didn't add this one and it already existed): https://gunmagwarehouse.com/kci-ar-15-223-5-56mm-50-round-drum-magazine.html
60-round STANAG: https://palmettostatearmory.com/ati-schmeisser-s60-223-5-56-ar-15-60-round-magazine-black.html
100-round double-drum STANAG: https://gunmagwarehouse.com/kci-ar-15-223-5-56-100-round-drum-magazine-with-personal-loader-and-pouch.html
100-round casket STANAG: https://www.surefire.com/mag5-100.html
150-round STANAG: https://www.drummagazines.com/SAW-MAG-150-Round-Drum-for-the-AR15-_p_150.html

**Comment by carmakazi** [comment]
For the 60-round mag, Surefire makes a 60-round version of that 100-round casket you linked.  It's probably more well known and a bit more reliable than the ATI mag, and perhaps more importantly, it includes dimensions and unloaded weight in the listing.

https://www.surefire.com/tactical-equipment/high-capacity-magazines/mag5-60.html

The Magpul D60 is also a quite popular and seemingly reliable 60 round drum magazine.

https://magpul.com/firearm-accessories/pmags/ar15-m4-m16/pmagd60ar-m4-5-56x45drum.html?mp_global_color=118

**Comment by carmakazi** [comment]
> The 100 round drum should be renamed to the Beta C. Mat type = plastic +steel.
> The 50, probably renam to the X-15, or something similarly popular
> The 150 could get renamed to the SAW mag? Mat type = plastic+steel.

I'd personally keep the item names as-is.  The Beta C company and X-products make their drums for several different guns and keeping the name would make it more clear what it is compatible with.  His adding the names to the description is a good middle ground I think.

**Comment by Rail-Runner** [comment]
Made changes are requested. And yeah, I decided to add specific names to descriptions instead; this is a minor detail for most players after all.

Also added 60-round drum magazine and tried to adjust weight to be a bit more accurate.

**Comment by kholat** [comment]
What about adding an adapter for g36 for stanags for instance?
https://www.tommybuilttactical.com/r-8-g36a
Would finally make use of the "magazine" mod slot I guess?

**Comment by Rail-Runner** [comment]
> What about adding an adapter for g36 for stanags for instance?
> https://www.tommybuilttactical.com/r-8-g36a
> Would finally make use of the "magazine" mod slot I guess?

It's worth adding, but seems outside of scope of this PR (and it's already fairly large). Maybe I'll add it later.

**Pull request merged**